### PR TITLE
drivers: defined charger driver api's.

### PR DESCRIPTION
**Pull request by rkotabag**
Defined API's header file for new charger driver.

Signed-off-by: Ramesh Kotabagi <ramesh.kotabagi@intel.com>

### CODE REVIEW

**Review by pabigot — CHANGES_REQUESTED** [review]
Same as #26974: It would be OK to add this as a specific driver for a specific charger management capability, but a generic API needs to be developed taking into account #2542.

There may also be some precedent for treating this functionality as a sensor, e.g. ti,bq274xx.

File: dts/bindings/charger/renesas,isl92xx.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2020 Intel Corporation
```
Comment (nashif) [inline dts/bindings/charger/renesas,isl92xx.yaml:1]: this should come with the driver it supports, should be separate from the API introduction/discussion.

File: drivers/charger/Kconfig [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Charger configuration options
+
```
Comment (nashif) [inline drivers/charger/Kconfig:2]: Kconfigs needs to be used and should come with the driver/subsystem they support.

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by pabigot** [comment]
See https://github.com/zephyrproject-rtos/zephyr/pull/26974#issuecomment-661311525 for further information on this approach.

File: CODEOWNERS [hunk 3]
```diff
@@ -144,6 +144,7 @@
 /drivers/bluetooth/                       @joerchan @jhedberg @Vudentz
 /drivers/can/                             @alexanderwachter
 /drivers/can/*mcp2515*                    @karstenkoenig
+/drivers/charger/                         @rgundi
```
Comment (carlescufi) [inline CODEOWNERS:147]: Is that correct? your current handle seems to be @rkotabag

**Comment by carlescufi** [comment]
@rkotabag are you planning to add an implementation of this driver? This is a requirement for any new API:

>The API introduction should be accompanied by at least one implementation of said API (in the case of peripheral APIs, this corresponds to one driver)

https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#experimental

**Comment by rkotabag** [comment]
> @rkotabag are you planning to add an implementation of this driver? This is a requirement for any new API:
> 
> > The API introduction should be accompanied by at least one implementation of said API (in the case of peripheral APIs, this corresponds to one driver)
> 
> https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#experimental

Up streaming the driver implementation for defined charger API's is currently dropped after discussing with our internal team, due to the following reasons,
1. Defined API and driver implementation are specific to particular chip and we are not sure w.r.t NDA and IP regarding same to open source the implementation.
2. Defined API's does not  meet 100% expectations of #2542 
So its decided internally that we will modify the current API for charger and battery driver to meet the requirement #2542  

So this pull request will be closed and will be working on to meet the requirement of #2542 .

**Comment by rkotabag** [comment]
Up streaming the driver implementation for defined charger API's is currently dropped after discussing with our internal team, due to the following reasons,
1. Defined API and driver implementation are specific to particular chip and we are not sure w.r.t NDA and IP regarding same to open source the implementation.
2.Defined API's does not meet 100% expectations of #2542

So its decided that we will be working on to meet the requirement of #2542  and close this pull request.

I am closing the this pull request as commented above. 
I thank everyone for there valuable inputs and exposure to #2542 requirement.

**Pull request closed without merging**